### DISCUSSION TITLE
Why there isn't an option to specify how to handle missing members in JsonSerializer.Deserialize

### DISCUSSION THREAD

**Discussion by shadow-cs** [dotnet/runtime#63736] [Q&A] [answered]
I'm wondering why `System.Text.Json` deserializer just ignores properties not present in the class being deserialized (unless `JsonExtensionData` is used) and there isn't an option to specify the behavior.

More info: <https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-6-0#missingmemberhandling>

This feature should be simple to implement by "just" adding a check to `ObjectDefaultConverter<T>.ReadPropertyValue` (well and possibly to some other places as well, maybe the generators, I'm not deep enough to say). I'd like to start using `S.T.Json` but this is a blocker to my use-case.

**Comment by ericstj** [comment, marked as answer]
Have a look here https://github.com/dotnet/runtime/issues/37483.  @eiriktsarpalis provided a workaround. https://github.com/dotnet/runtime/issues/37483#issuecomment-950774100

**Reply by shadow-cs** [reply]
Thanks for your response (and helping me find the issue, I got the keywords wrong and was unable to find it), I was initially thinking about using `JsonExtensionData`, but in the end, I decided this isn't a dealbreaker for our use case. We should be able to detect most misuses during testing which incorporates JSON Schema validation.

But this seems to be a simple enough (I hope) feature to implement that it feels unnatural to use a `Dictionary` and pollute the classes.

I hope the issue gets some traction...
